StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1113CommaMustBeOnSameLineAsPreviousParameter.cs

```diff
@@ -88,361 +88,239 @@ private void HandleArrayDeclaration(SyntaxNodeAnalysisContext context)
-
-                var commaLocation = comma.GetLocation();
-                if (commaLocation.GetLineSpan().StartLinePosition.Line !=
-                    previousParameter.GetLocation().GetLineSpan().EndLinePosition.Line)
+                var parameters = parameterList.Parameters;
+                if (parameters.Count > 1)
                 {
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, commaLocation));
+                    CheckIfCommasAreAtTheSameLineAsThePreviousParameter(context, parameters.GetWithSeparators());
                 }
             }
         }
 
-        private static void CheckIfCommasAreAtTheSameLineAsThePreviousParameter(SyntaxNodeAnalysisContext context, List<SyntaxToken> commas,
-            AttributeListSyntax attributeListSyntax)
+        private static void CheckIfCommasAreAtTheSameLineAsThePreviousParameter(SyntaxNodeAnalysisContext context, SyntaxNodeOrTokenList nodeOrTokenList)
         {
-            for (int index = 0; index < commas.Count; index++)
+            // If flag is even we expecting parameter syntax node, otherwise we expecting comma token.
+            int index = 0;
+            SyntaxNode previousNode = null;
+
+            foreach (var nodeOrToken in nodeOrTokenList)
             {
-                var comma = commas[index];
-                if (attributeListSyntax.Attributes.Count <= index)
+                if (index % 2 == 0)
                 {
-                    return;
-                }
-
-                var previousParameter = attributeListSyntax.Attributes[index];
+                    // We expecting node here
+                    if (nodeOrToken.IsToken)
+                    {
+                        return;
+                    }
 
-                var commaLocation = comma.GetLocation();
-                if (commaLocation.GetLineSpan().StartLinePosition.Line !=
-                    previousParameter.GetLocation().GetLineSpan().EndLinePosition.Line)
-                {
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, commaLocation));
+                    previousNode = nodeOrToken.AsNode();
                 }
-            }
-        }
-
-        private static void CheckIfCommasAreAtTheSameLineAsThePreviousParameter(SyntaxNodeAnalysisContext context, List<SyntaxToken> commas, ArrayRankSpecifierSyntax arrayRankSpecifierSyntax)
-        {
-            for (int index = 0; index < commas.Count; index++)
-            {
-                var comma = commas[index];
-                if (arrayRankSpecifierSyntax.Sizes.Count <= index)
+                else
                 {
-                    return;
+                    // We expecting token here
+                    if (nodeOrToken.IsNode)
+                    {
+                        return;
+                    }
+
+                    // previousNode not set
+                    if (previousNode == null)
+                    {
+                        return;
+                    }
```

Comment:
This check is superfluous. It will never trigger, unless there is a bug in Roslyn.